craft_parts/executor/collisions.py

```diff
@@ -100,8 +123,8 @@ def paths_collide(path1: str, path2: str) -> bool:
     if not (path1_is_dir and path2_is_dir) and _file_collides(path1, path2):
         return True
 
-    # Otherwise, paths do not conflict.
-    return False
+    # Otherwise, paths only conflict if both have permissions defined for them.
+    return all([permissions_path1, permissions_path2])
```

Comment:
Shouldn't they conflict if the permissions are actually different?

Reply:
As we discussed, I pushed a commit that does some comparing logic to have paths collide if their permissions are "different". The code is probably unorganized and overcomplicated because my brain is a bit fried, but please take a look. Maybe you have a cleaner approach.
The failing test is, as far as I can tell, unrelated to this PR (a `git` update broke a test)